parser/src/main/kotlin/com/linkedin/dex/parser/JUnit4Extensions.kt

```diff
@@ -119,20 +119,39 @@ private fun createAllTestMethods(
             val childClassAnnotations = dexFile.getClassAnnotationValues(directory)
             val childClassAnnotationNames = childClassAnnotations.map { it.name }
 
+            // We need to differentiate cases where a method is overridden from the superclass
+            // vs when they are not, as it will impact whether we should include all annotations
+            // from the superclass method or just ones with the inherited property
+            // So, we exclude any that have overridden versions
             val adaptedSuperMethods = superTestMethods
+                .filterNot { superMethod -> parsingResult.testMethods.any {
+                    it.testNameWithoutClass.equals(superMethod.testNameWithoutClass) } }
```

Comment:
I think you should theoretically have an IDE warning about using `.equals()` here
```suggestion
                .filterNot { superMethod -> parsingResult.testMethods.any {
                    it.testNameWithoutClass == superMethod.testNameWithoutClass
                 } }
```